src/lib/Shapes.ts

```diff
@@ -101,3 +106,7 @@ export class Shapes {
 		return new MapValidator(keyValidator, valueValidator);
 	}
 }
+
+type UnwrapTuple<T extends [...any[]]> = T extends [infer Head, ...infer Tail]
```

Comment:
Shouldn't we export this? Might be useful for some people 👀

Reply:
Why... would you do that? I mean, it's an *extremely specific* recursive type unwrapper for Shapeshift